streamwebs_frontend/streamwebs/views.py

```diff
@@ -244,3 +246,66 @@ def water_quality_edit(request, site_slug):
     return render(request, 'streamwebs/datasheets/water_quality_edit.html', {
         'site': site
     })
+
+
+def riparian_transect_view(request, site_slug, data_id):
+    transects = RiparianTransect.objects.filter(site_id=site_slug)
+    transect = transects.get(id=data_id)
+    zones = TransectZone.objects.filter(transect_id=transect)
+    site = Site.objects.get(id=site_slug)
+
+    # Invoking the database by evaluating the queryset before passing it to the
+    # template is necessary in order to pass Travis tests.
+    # https://docs.djangoproject.com/en/1.9/ref/models/querysets/#when-querysets-are-evaluated
+    zones = list(zones)
+
+    return render(
+        request, 'streamwebs/datasheets/riparian_transect_view.html', {
+            'transect': transect,
+            'zones': zones,
+            'site': site
+            }
+        )
+
+
+def riparian_transect_edit(request, site_slug):
+    """
+    The view for the submission of a new riparian transect data sheet.
+    """
+    added = False
+    site = Site.objects.get(id=site_slug)
```

Comment:
(line 276 and others) If the id field is number, and site_slug is a string, how does this work?  I must be missing some django magic or something.